templates/boltzmann/middleware/honeycomb.ts

```diff
@@ -103,24 +158,137 @@ function trace ({
   }
 }
 
-function honeycombMiddlewareSpans ({name}: {name?: string} = {}) {
-  if (!process.env.HONEYCOMB_WRITEKEY) {
-    return (next: Handler) => (context: Context) => next(context)
-  }
-
+function beelineMiddlewareSpans ({name}: {name?: string} = {}) {
   return function honeycombSpan (next: Handler) {
     return async (context: Context) => {
       const span = beeline.startSpan({
-        name: `mw: ${name}`
+        name: middlewareSpanName(name)
+      })
+
+      beeline.addContext({
+        [otelSemanticConventions.SemanticResourceAttributes.SERVICE_NAME]: honeycomb.options.serviceName,
+        'honeycomb.trace_type': 'beeline'
       })
 
       // Assumption: the invariant middleware between each layer
       // will ensure that no errors are thrown from next().
       const result = await next(context)
+
       beeline.finishSpan(span)
       return result
     }
   }
 }
 
+/*
+ * ┏┓
+ * ┃┃╱╲ in
+ * ┃╱╱╲╲ this
+ * ╱╱╭╮╲╲house
+ * ▔▏┗┛▕▔ we
+ * ╱▔▔▔▔▔▔▔▔▔▔╲
+ * trace with opentelemetry
+ * ╱╱┏┳┓╭╮┏┳┓ ╲╲
+ * ▔▏┗┻┛┃┃┗┻┛▕▔
+ */
+function otelTrace () {
+  return function honeycombTrace (next: Handler) {
+    return (context: Context) => {
+      let traceContext = otel.context.active()
+
+      // Typically, HTTP auto-instrumentation will create a parent span for us
+      let span: otel.Span | undefined = otel.trace.getSpan(traceContext)
+
+      // startSpan should always return a span, but as far as typescript is
+      // concerned the span could still be undefined. We could assert that it
+      // exists, but throwing instrumentation-related errors is poor form.
+      if (span) {
+        span.setAttribute('boltzmann.http.query', context.url.search)
+
+        span.updateName(traceName(context.method, context.url.pathname))
+
+        traceContext = otel.trace.setSpan(traceContext, span)
+
+        if (isDev()) {
+          context._honeycombTrace = span
+        }
+      } else {
+        otel.diag.error(String(new assert.AssertionError({
+          message: 'no parent span found or created',
+          actual: span,
+          expected: true,
+          operator: '=='
+        }).stack))
+      }
+
+      onHeaders(context._response, function () {
+        const handler: Handler = <Handler>context.handler
+
+        if (!span) {
+          return
+        }
+
+        span.setAttribute(
+          otelSemanticConventions.SemanticResourceAttributes.SERVICE_VERSION,
+          <string>handler.version
+        )
+
+        Object.entries(context.params).forEach(([key, value]) => {
```

Comment:
I looked closer at this, and yes that's true, *however* I'm actually doing a map across the keys on L238 (`paramSpanAttribute(key)`). I could do an `Object.fromEntries` and pass *that* to `span.setAttributes` but I don't feel like it would make this more clear. But maybe I'm missing something?

Reply:
I believe this is ready to resolve after my investigation :point_up: but I'll keep it open so we can ensure consensus on this one.